eq-author-api/src/validation/customKeywords/validateSurveyId.js

```diff
@@ -19,24 +21,56 @@ module.exports = (ajv) =>
       // Get the supplementary data from the questionnaire object
       const supplementaryData = questionnaire.supplementaryData;
 
-      // If supplementaryData exists and contains a surveyId, and supplementaryData's surveyId doesn't match the questionnaire's surveyId, throw a validation error
       if (
-        supplementaryData &&
-        supplementaryData.surveyId &&
-        questionnaireSurveyId !== supplementaryData.surveyId
+        typeof questionnaire.surveyId === "string" &&
+        questionnaire.surveyId.length === 0
       ) {
         isValid.errors = [
           createValidationError(
             instancePath,
             fieldName,
-            ERR_SURVEY_ID_MISMATCH,
+            ERR_VALID_REQUIRED,
             questionnaire,
-            ERR_SURVEY_ID_MISMATCH
+            ERR_VALID_REQUIRED
           ),
         ];
         return false;
-      }
+      } else {
```

Comment:
Ideally, it must be like this - if (empty) else if (invalid format) else (supplementary data mismatch)